CMakeLists.txt

```diff
@@ -111,105 +102,78 @@ if(CMAKE_GENERATOR STREQUAL "Unix Makefiles")
-                -DTUTORIALS_DATA_OUTPUT_DIRECTORY:STRING=${TUTORIALS_DATA_OUTPUT_DIRECTORY}
-                -DBUILD_DOCUMENTATION:BOOL=${BUILD_DOCUMENTATION}
-                -DBUILD_GEOGRAM_WITHOUT_EXE:BOOL=${BUILD_GEOGRAM_WITHOUT_EXE}
+                -G ${CMAKE_GENERATOR}
+                ${RINGMESH_EXTRA_ARGS} 
+                -DCMAKE_BUILD_TYPE=${config}
+                -DGLOBAL_BINARY_DIR=${PROJECT_BINARY_DIR}
                 WORKING_DIRECTORY ${project_binary_dir_config})
       endforeach()
 
       # Get out when all configurations have been run
       # We do not want to execute the rest of the file in that case
       return()
     endif()
-else(CMAKE_GENERATOR STREQUAL "Unix Makefiles")
-    set(GLOBAL_BINARY_DIR  ${PROJECT_BINARY_DIR})
 endif(CMAKE_GENERATOR STREQUAL "Unix Makefiles")
 
+if(USE_SUPERBUILD)
+    # execute the superbuild
+    project(SUPERBUILD)
+    if(NOT GLOBAL_BINARY_DIR)
+        set(GLOBAL_BINARY_DIR ${PROJECT_BINARY_DIR})
+    endif()
+    include("${CMAKE_SOURCE_DIR}/cmake/SuperBuild.cmake")
+    return()
+endif()
+
+# Define the project RINGMesh
+project(RINGMesh)
+
+#------------------------------------------------------------------------------------------------
+# Turn on the ability to create folders to organize projects and files
+# It creates "CMakePredefinedTargets" folder by default and adds CMake
+# defined projects like INSTALL.vcproj and ZERO_CHECK.vcproj
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+
+# Define version number
+# It is then exported to the configuration file
+set(RINGMesh_VERSION_MAJOR 5)
+set(RINGMesh_VERSION_MINOR 0)
+set(RINGMesh_VERSION_PATCH 0)
+set(RINGMesh_VERSION ${RINGMesh_VERSION_MAJOR}.${RINGMesh_VERSION_MINOR}.${RINGMesh_VERSION_PATCH})
+
 message(STATUS "RINGMesh binary directory is: ${PROJECT_BINARY_DIR}")
 message(STATUS "RINGMesh source directory is: ${PROJECT_SOURCE_DIR}")
 
 include(cmake/utils.cmake)
 
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+    
 #------------------------------------------------------------------------------------------------
 # Platform dependent settings
 if(UNIX)
-    set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
     add_compile_options(-Wall -Wextra -Wno-long-long -Wconversion
         -Wsign-conversion -Wdouble-promotion -Wno-attributes)
 
-    # Determine gcc version and activate additional warnings available in latest versions
-    execute_process(COMMAND ${CMAKE_CXX_COMPILER} -dumpversion OUTPUT_VARIABLE COMPILER_VERSION)
-
-    # C++11 fully works on Linux from gcc/g++ 4.8 (before not fully operational).
-    # On MacOS, it seems that C++11 is working since gcc/g++ 4.2.
     if(APPLE)
         if (NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-            message(FATAL_ERROR "RINGMesh on Apple requires Clang compiler")
+            message(WARNING "RINGMesh on Apple is only tested with Clang compiler")
```

Comment:
Should be keep this message? Maybe it is working with something else, way blocking? Not even sure about the warning..